packages/cli/src/cli/cogify/batch.job.ts

```diff
@@ -29,34 +30,41 @@ export class BatchJob {
    *
    * This needs to be within `[a-Z_-]` upto 128 characters log
    * @param job job to process
-   * @param fileName output filename
+   * @param fileNames output filename
    * @returns job id
    */
-  static id(job: CogJob, fileName: string): string {
+  static id(job: CogJob, fileNames: string[]): string {
     // Job names are uncontrolled so hash the name and grab a small slice to use as a identifier
     const jobName = createHash('sha256').update(job.name).digest('hex').slice(0, 16);
-    return `${job.id}-${jobName}-${fileName}`.slice(0, 128);
+    fileNames.sort((a, b) => (a > b ? 1 : -1));
+    return `${job.id}-${jobName}-${fileNames.join('_')}`.slice(0, 128);
   }
 
   static async batchOne(
     jobPath: string,
     job: CogJob,
     batch: Batch,
-    name: string,
+    names: string[],
     isCommit: boolean,
   ): Promise<{ jobName: string; jobId: string; memory: number }> {
-    const jobName = BatchJob.id(job, name);
-    const tile = TileMatrixSet.nameToTile(name);
-    const alignmentLevels = Projection.findAlignmentLevels(job.tileMatrix, tile, job.source.gsd);
-    // Give 25% more memory to larger jobs
-    const resDiff = 1 + Math.max(alignmentLevels - MagicAlignmentLevel, 0) * 0.25;
-    const memory = 3900 * resDiff;
+    const jobName = BatchJob.id(job, names);
+
+    // Calculate the total size to provision memory
+    let res = 1;
+    for (const name of names) {
+      const tile = TileMatrixSet.nameToTile(name);
+      const alignmentLevels = Projection.findAlignmentLevels(job.tileMatrix, tile, job.source.gsd);
+      // Give 25% more memory to larger jobs
+      res += Math.max(alignmentLevels - MagicAlignmentLevel, 0) * 0.25;
```

Comment:
is this adding a number per names? what is memory for a job with 3 images?

Reply:
I am not 100% sure the logic here. I think we might only need to do this for the `names.lenght === 1`, as others just small images chunk job.